src/upgrade.py

```diff
@@ -120,6 +141,25 @@ def _on_upgrade_changed(self, _) -> None:
 
         self.charm.update_config()
 
+    def _on_upgrade_charm_check_legacy(self, event: UpgradeCharmEvent) -> None:
+        if not self.peer_relation:
+            logger.debug("Wait all units join the upgrade relation")
+            return
+
+        if self.state:
+            # Do nothing - if state set, upgrade is supported
+            return
+
+        logger.warning("Upgrading from unsupported version")
```

Comment:
```suggestion
        logger.warning("Upgrading from unspecified version")
```
Isn't this for current stable?

Reply:
Exactly. Fixed on https://github.com/canonical/postgresql-k8s-operator/pull/210/commits/7c8bed563bf4ad55fb64ed316cf50ce26c1069f6.